services/requester/requester.go

```diff
@@ -289,44 +292,65 @@ func (e *EVM) GetNonce(ctx context.Context, address common.Address) (uint64, err
 }
 
 func (e *EVM) Call(ctx context.Context, address common.Address, data []byte) ([]byte, error) {
+	hexEncodedData, err := cadence.NewString(hex.EncodeToString(data))
+	if err != nil {
+		return nil, err
+	}
+
 	// todo make "to" address optional, this can be used for contract deployment simulations
-	txData := cadenceArrayFromBytes(data).WithType(byteArrayType)
-	toAddress := cadenceArrayFromBytes(address.Bytes()).WithType(addressType)
+	hexEncodedAddress, err := cadence.NewString(address.Hex()[2:])
```

Comment:
I would prefer: `strings.TrimPrefix(address.Hex(), "0x")` It's a bit clearer what it does.